tools/swift-demangle/swift-demangle.cpp

```diff
@@ -175,7 +175,7 @@ static int demangleSTDIN(const swift::Demangle::DemangleOptions &options) {
   // This doesn't handle Unicode symbols, but maybe that's okay.
   // Also accept the future mangling prefix.
   // TODO: remove the "_S" as soon as MANGLING_PREFIX_STR gets "_S".
-  llvm::Regex maybeSymbol("(_T|_*\\$S|" MANGLING_PREFIX_STR ")[_a-zA-Z0-9$.]+");
+  llvm::Regex maybeSymbol("_(T|\\$[Ss]|" MANGLING_PREFIX_STR ")[_a-zA-Z0-9$.]+");
```

Comment:
@graydon and I just noticed that `MANGLING_PREFIX_STR` will have an unescaped `$` in it, which means that things *without* the leading underscore weren't getting matched. Moving the underscore to the outside doesn't fix that either.

Reply:
Maybe `"(_T|_?\\$[Ss])"` and leave out the `MANGLING_PREFIX_STR` part entirely?

Reply:
Well, you didn't regress it. :-)